common/src/main/java/io/netty/util/concurrent/UnorderedThreadPoolEventExecutor.java

```diff
@@ -0,0 +1,256 @@
+     * using {@link DefaultThreadFactory}.
+     */
+    public UnorderedThreadPoolEventExecutor(int corePoolSize) {
+        this(corePoolSize, new DefaultThreadFactory(UnorderedThreadPoolEventExecutor.class));
+    }
+
+    /**
+     * See {@link ScheduledThreadPoolExecutor#ScheduledThreadPoolExecutor(int, ThreadFactory)}
+     */
+    public UnorderedThreadPoolEventExecutor(int corePoolSize, ThreadFactory threadFactory) {
+        super(corePoolSize, threadFactory);
+    }
+
+    /**
+     * Calls {@link UnorderedThreadPoolEventExecutor#UnorderedThreadPoolEventExecutor(int,
+     * ThreadFactory, java.util.concurrent.RejectedExecutionHandler)} using {@link DefaultThreadFactory}.
+     */
+    public UnorderedThreadPoolEventExecutor(int corePoolSize, java.util.concurrent.RejectedExecutionHandler handler) {
+        this(corePoolSize, new DefaultThreadFactory(UnorderedThreadPoolEventExecutor.class), handler);
+    }
+
+    /**
+     * See {@link ScheduledThreadPoolExecutor#ScheduledThreadPoolExecutor(int, ThreadFactory, RejectedExecutionHandler)}
+     */
+    public UnorderedThreadPoolEventExecutor(int corePoolSize, ThreadFactory threadFactory,
+                                            java.util.concurrent.RejectedExecutionHandler handler) {
+        super(corePoolSize, threadFactory, handler);
+    }
+
+    @Override
+    public EventExecutor next() {
+        return this;
+    }
+
+    @Override
+    public EventExecutorGroup parent() {
+        return this;
+    }
+
+    @Override
+    public boolean inEventLoop() {
+        return false;
+    }
+
+    @Override
+    public boolean inEventLoop(Thread thread) {
+        return false;
+    }
+
+    @Override
+    public <V> Promise<V> newPromise() {
+        return new DefaultPromise<V>(this);
+    }
+
+    @Override
+    public <V> ProgressivePromise<V> newProgressivePromise() {
+        return new DefaultProgressivePromise<V>(this);
+    }
+
+    @Override
+    public <V> Future<V> newSucceededFuture(V result) {
+        return new SucceededFuture<V>(this, result);
+    }
+
+    @Override
+    public <V> Future<V> newFailedFuture(Throwable cause) {
+        return new FailedFuture<V>(this, cause);
+    }
+
+    @Override
+    public boolean isShuttingDown() {
+        return isShutdown();
+    }
+
+    @Override
+    public List<Runnable> shutdownNow() {
+        List<Runnable> tasks = super.shutdownNow();
+        terminationFuture.trySuccess(null);
+        return tasks;
+    }
+
+    @Override
+    public void shutdown() {
+        super.shutdown();
+        terminationFuture.trySuccess(null);
+    }
+
+    @Override
+    public Future<?> shutdownGracefully() {
+        return shutdownGracefully(2, 15, TimeUnit.SECONDS);
+    }
+
+    @Override
+    public Future<?> shutdownGracefully(long quietPeriod, long timeout, TimeUnit unit) {
+        // TODO: At the moment this just calls shutdown but we may be able to do something more smart here which
```

Comment:
![INFO](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-info.png) Complete the task associated to this TODO comment. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://garage.netty.io/sonarqube/coding_rules#rule_key=squid%3AS1135)